dask/array/core.py

```diff
@@ -2594,8 +2598,9 @@ def from_array(
         If ``x`` doesn't support concurrent reads then provide a lock here, or
         pass in True to have dask.array create one for you.
     asarray : bool, optional
-        If True (default), then chunks will be converted to instances of
-        ``ndarray``. Set to False to pass passed chunks through unchanged.
+        If True then call np.asarray on chunks to convert them to numpy arrays.
+        Set to False to pass passed chunks through unchanged.  This defaults to
+        True, unless the input has the ``__array_function__`` method defined.
```

Comment:
This is a bit confusing. Options are `True` or `False`, but the actual default value is `None` and description says it defaults to `True`. I would either explain the three options or just have `asarray=True` as default and use `__array_function__` if available.

Reply:
I've pushed up some new language if you want to take a look.